src/main/java/com/gluonhq/substrate/config/ConfigResolver.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2019, Gluon
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.substrate.config;
+
+import com.gluonhq.substrate.model.ClassPath;
+import com.gluonhq.substrate.util.FileOps;
+import com.gluonhq.substrate.util.Logger;
+import com.gluonhq.substrate.util.Strings;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import static com.gluonhq.substrate.Constants.USER_INIT_BUILD_TIME_ARCHOS_FILE;
+import static com.gluonhq.substrate.Constants.USER_INIT_BUILD_TIME_FILE;
+import static com.gluonhq.substrate.Constants.META_INF_SUBSTRATE_CONFIG;
+import static com.gluonhq.substrate.Constants.USER_JNI_ARCHOS_FILE;
+import static com.gluonhq.substrate.Constants.USER_JNI_FILE;
+import static com.gluonhq.substrate.Constants.USER_REFLECTION_ARCHOS_FILE;
+import static com.gluonhq.substrate.Constants.USER_REFLECTION_FILE;
+
+/**
+ * Helper class that helps scanning jars in the classpath looking for
+ * files in META-INF/substrate/config that contain information that has
+ * to be added to the config files and native image command line flags
+ */
+public class ConfigResolver {
+
+    private final List<File> jars;
+
+    public ConfigResolver(String classpath) {
+        this.jars = new ClassPath(classpath)
```

Comment:
Since we require class path as a constructor parameter, why not to use `ClassPath`? 
I suspect that that is what the caller would have anyway?

Reply:
Right, but this probably leads to a bigger refactoring. We have `processClassPath`, that returns a String, while it probably could return a `ClassPath` object, and that will be the only object we would need to create:
```
public boolean compile(String cp) throws IOException, InterruptedException {
        ClassPath classPath = processClassPath(cp);
    ...
```

Does that make sense? Probably for a follow-up PR then.